src/main/java/org/opensearch/jobscheduler/utils/JobDetailsService.java

```diff
@@ -0,0 +1,309 @@
+     * @param listener an {@code ActionListener} that has onResponse and onFailure that is used to return the job details if it was created
+     *                 or else null.
+     */
+    private void createJobDetailsForExtensionId(final JobDetails tempJobDetails, String extensionId, ActionListener<JobDetails> listener) {
+        try {
+            final IndexRequest request = new IndexRequest(JOB_DETAILS_INDEX_NAME).id(extensionId)
+                .source(tempJobDetails.toXContent(XContentFactory.jsonBuilder(), ToXContent.EMPTY_PARAMS))
+                .setIfSeqNo(SequenceNumbers.UNASSIGNED_SEQ_NO)
+                .setIfPrimaryTerm(SequenceNumbers.UNASSIGNED_PRIMARY_TERM)
+                .create(true);
+            client.index(request, ActionListener.wrap(response -> listener.onResponse(new JobDetails(tempJobDetails)), exception -> {
+                if (exception instanceof VersionConflictEngineException) {
+                    logger.debug("Job Details for extension id " + extensionId + " is already created. {}", exception.getMessage());
+                }
+                if (exception instanceof IOException) {
+                    logger.error("IOException occurred creating job details", exception);
+                }
+                listener.onResponse(null);
+            }));
+        } catch (IOException e) {
+            logger.error("IOException occurred creating job details for extension id " + extensionId, e);
+            listener.onResponse(null);
+        }
+    }
+
+    /**
+     *
+     * @param extensionId unique id to find the job details document in the index
+     * @param listener an {@code ActionListener} that has onResponse and onFailure that is used to return the job details if it was found
+     *                 or else null.
+     */
+    private void findJobDetailsForExtensionId(final String extensionId, ActionListener<JobDetails> listener) {
+        GetRequest getRequest = new GetRequest(JOB_DETAILS_INDEX_NAME).id(extensionId);
+        client.get(getRequest, ActionListener.wrap(response -> {
+            if (!response.isExists()) {
+                listener.onResponse(null);
+            } else {
+                try {
+                    XContentParser parser = XContentType.JSON.xContent()
+                        .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, response.getSourceAsString());
+                    parser.nextToken();
+                    listener.onResponse(JobDetails.parse(parser));
+                } catch (IOException e) {
+                    logger.error("IOException occurred finding JobDetails for extension id " + extensionId, e);
+                    listener.onResponse(null);
+                }
+            }
+        }, exception -> {
+            logger.error("Exception occurred finding job details for extension id " + extensionId, exception);
+            listener.onFailure(exception);
+        }));
+    }
+
+    /**
+     *
```

Comment:
Need a description of what the method actually does.

Reply:
done